woocommerce/woocommerce-ios
# [Mobile Payments] Merge In Person Payments into develop

Closes #3984

This PR contains the implementation of In-Person Payments. The goal of this PR is not testing the feature in itself (there will be a separate call for testing later) but **to make sure that the feature can be dark-deployed safely** (more about what that means in the testing instructions)

We submit this now because we want to get into the 6.8 code-freeze, because:
* We could use the 6.8 submission to test if Apple will react the build due to the way we have integrated the Stripe SDK
* We have time to undo the merge, if needed
* We can issue a call for testing after the feature has landed in develop
* We start preparing for the actual submission on June 25 with enough time to iron out quirks. 

**The feature is gated by**:
* a local feature flag (`cardPresentPayments`) that is only true for localDeveloper and alpha flags
* a flag in WCPayAccount that opens the feature to specific stores gated into it (see p91TBi-5cf-p2)

Following @astralbodies suggestion, we have a contingency plan to undo the merge if Apple rejects our build for who knows why. (#4253 that will be accompanied by a checklist in the form of a p2 post)

## Changes
* Implementation of In-person Payments.
* Integration with Stripe Terminal SDK. Which requires adding new keys to the Info.pist file.
* New UI to discover and connect to card readers. 
* New UI to take card present payments.

## How to test
As it is, the feature is hidden behind a local feature flag and a remote flag in the WCPayAccount object.

The Stripe Terminal SDK requires access to location data and to bluetooth. **We want to make sure that users that are not allowed to see the feature do not get prompted by the system for permissions to access bluetooth or location data**. Our goal is deploying the feature in 6.8, without users noticing that it's there.

Test 1 (site without WCPay installed): 
1. Checkout the branch. Run `bundle exec pod install`
2. Build and run to a device or simulator where the feature has never been run before, ideally in  We want to do this so that we can check that the app does not prompt for location or bluetooth permissions.
3. Log into the app. Ideally, with a store/credentials that **do not have** WCPay installed.
4. Navigate to Settings. There shouldn't be any entry related to Card Readers (i.e. "Manage Card Readers")
5. Navigate to an order with payment method Cash on Delivery. There should not be any button to "Collect Payment".
6. Navigate around the app (try loading orders, navigate to notifications)
7. Kill the app, start again, try the same steps
8. **The app should not prompt for location or bluetooth permissions anytime.**

Test 2 (site with WCPay installed but not gated into beta 1):
1. Checkout the branch. Run `bundle exec pod install`
2. Build and run to a device or simulator where the feature has never been run before, ideally in  We want to do this so that we can check that the app does not prompt for location or bluetooth permissions.
3. Log into the app. Ideally, with a store/credentials that **do have** WCPay installed.
4. Navigate to Settings. There shouldn't be any entry related to Card Readers (i.e. "Manage Card Readers")
5. Navigate to an order with payment method Cash on Delivery. There should not be any button to "Collect Payment".
6. Navigate around the app (try loading orders, navigate to notifications)
7. Kill the app, start again, try the same steps
8. **The app should not prompt for location or bluetooth permissions anytime.**

Notice the pattern: we want to make sure that the app does not request access to location or bluetooth.

cc @AliSoftware . Once we merge this, we might need to update the privacy recipe in the App Store listing.

Update release notes:
